azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -148,7 +148,7 @@ private void logRequest(final HttpServletRequest req, final Session session) {
     buf.append("\"");
     buf.append(req.getMethod()).append(" ");
     buf.append(req.getRequestURI()).append(" ");
-    if (req.getQueryString() != null) {
+    if (req.getQueryString() != null && allowedPostRequest(req)) {
```

Comment:
A && !(A && B && C)
--> A && (!A || !B || !C) de Morgan's law
--> (A && !A) || (A && !B) || (A && !C) distributive law
--> (A && !B) || (A && !C) because (A && !A) can never happen
--> A && (!B || !C) distributive law in reverse
And A && (!B || !C) != !(A && B && C) because 1, 0, 0 leads to the first being true and the second being false.
So we should need both. Does that make sense? (Also, yay boolean algebra! :smile:)